code/modules/projectiles/guns/projectile/automatic.dm

```diff
@@ -270,5 +276,99 @@
+	name = "\improper Uzi"
+	desc = "A lightweight, compact, fast firing gun, for when you want someone really dead. Uses .45 rounds."
+	icon_state = "mini-uzi"
+	w_class = 3
+	load_method = MAGAZINE
+	caliber = ".45"
+	origin_tech = list(TECH_COMBAT = 5, TECH_MATERIAL = 2, TECH_ILLEGAL = 5)
+	magazine_type = /obj/item/ammo_magazine/c45uzi
+	allowed_magazines = list(/obj/item/ammo_magazine/c45uzi)
+
+	firemodes = list(
+		list(mode_name="semiauto", burst=1, fire_delay=0),
+		list(mode_name="short bursts", burst=4, burst_delay=1, fire_delay=3, move_delay=3, burst_accuracy = list(0,-1,-1,-2), dispersion = list(0.6, 1.0, 1.0, 1.2))
+		)
+
+/obj/item/weapon/gun/projectile/automatic/mini_uzi/update_icon()
+	..()
+	if(ammo_magazine)
+		icon_state = "mini-uzi"
+	else
+		icon_state = "mini-uzi-empty"
+
+/obj/item/weapon/gun/projectile/automatic/p90
+	name = "\improper H90K"
+	desc = "A compact, high capacity SMG produced by Hephaistos Industries. Despite its fierce reputation, it still manages to feel like a toy. Uses 5mm rounds."
+	icon_state = "p90smg"
+	item_state = "p90"
+	w_class = 3
+	caliber = "5mm"
+	origin_tech = list(TECH_COMBAT = 5, TECH_MATERIAL = 2)
+	slot_flags = SLOT_BELT // ToDo: Belt sprite.
+	fire_sound = 'sound/weapons/Gunshot_light.ogg'
+	load_method = MAGAZINE
+	magazine_type = /obj/item/ammo_magazine/c5mmt
+	allowed_magazines = list(/obj/item/ammo_magazine/c5mmt)
+
+	firemodes = list(
+		list(mode_name="semiauto", burst=1, fire_delay=0),
+		list(mode_name="short bursts", burst=5, burst_delay=1, fire_delay=4, move_delay=4, burst_accuracy = list(0,-1,-1,-2,-2), dispersion = list(0.6, 1.0, 1.0, 1.2, 1.2)),
+		)
 
+/obj/item/weapon/gun/projectile/automatic/p90/update_icon()
+	icon_state = "p90smg-[ammo_magazine ? round(ammo_magazine.stored_ammo.len, 6) : "empty"]"
 
+/obj/item/weapon/gun/projectile/automatic/tommygun
+	name = "\improper Tommygun"
+	desc = "Some little typewriter, huh? I'm gonna write my name all over this town with big letters! Uses .45 rounds."
+	icon_state = "tommygun"
+	w_class = 3
+	caliber = ".45"
+	origin_tech = list(TECH_COMBAT = 5, TECH_MATERIAL = 2, TECH_ILLEGAL = 5)
+	slot_flags = SLOT_BELT // ToDo: Belt sprite.
+	load_method = MAGAZINE
+	magazine_type = /obj/item/ammo_magazine/tommymag
+	allowed_magazines = list(/obj/item/ammo_magazine/tommymag, /obj/item/ammo_magazine/tommydrum)
+
+	firemodes = list(
+		list(mode_name="semiauto", burst=1, fire_delay=0),
+		list(mode_name="short bursts",	burst=5, move_delay=6, burst_accuracy = list(0,-1,-1,-2,-2), dispersion = list(0.6, 1.0, 1.0, 1.0, 1.2)),
+		)
+
+/obj/item/weapon/gun/projectile/automatic/tommygun/update_icon()
+	..()
+	icon_state = (ammo_magazine)? "tommygun" : "tommygun-empty"
+//	update_held_icon()
+
+/obj/item/weapon/gun/projectile/automatic/wt3000
+	name = "\improper WT3000"
+	desc = "The bullpup configured WT3000 is a light, compact, military-grade assault rifle that is most <i>definitely</i> illegal for civilians to own. What fun! Uses 5.56mm rounds."

```

Comment:
The description ceases to be interesting without that, and I don't want the descriptions to be dull and uninteresting, because really they only exist for fluff. There's no mechanical advantage to them, there's no explaining how the gun works, the only thing outright useful in the description is what kind of bullet it uses.

If I'm going to remove that, I need something interesting to replace it with.